include/engine/data_watchdog.hpp

```diff
@@ -26,10 +27,23 @@ class DataWatchdog
 {
   public:
     DataWatchdog()
-        : shared_barriers{std::make_shared<storage::SharedBarriers>()},
-          shared_regions(storage::makeSharedMemory(storage::CURRENT_REGION)),
-          current_timestamp{storage::REGION_NONE, 0}
+        : active(true)
+        , timestamp(0)
     {
+        watcher = std::thread(&DataWatchdog::Run, this);
+        while (!facade)
```

Comment:
I would like the class to be in a usable state before the constructor finishes.
Having a watchdog that can potentially return an empty `shared_ptr` will crash the user-code to crash.

You think it is okay to trade a heavy constructor for ease-of-use?